coordinator/tributary/src/blockchain.rs

```diff
@@ -111,32 +94,20 @@ impl<D: Db, T: TransactionTrait> Blockchain<D, T> {
   }
 
   pub(crate) fn block_from_db(db: &D, genesis: [u8; 32], block: &[u8; 32]) -> Option<Block<T>> {
-    db.get(Self::block_key(&genesis, block))
+    BlockDb::get(db, genesis, block)
       .map(|bytes| Block::<T>::read::<&[u8]>(&mut bytes.as_ref()).unwrap())
   }
 
-  pub(crate) fn commit_from_db(db: &D, genesis: [u8; 32], block: &[u8; 32]) -> Option<Vec<u8>> {
-    db.get(Self::commit_key(&genesis, block))
-  }
-
-  pub(crate) fn block_hash_from_db(db: &D, genesis: [u8; 32], block: u32) -> Option<[u8; 32]> {
-    db.get(Self::block_hash_key(&genesis, block)).map(|h| h.try_into().unwrap())
-  }
-
   pub(crate) fn commit(&self, block: &[u8; 32]) -> Option<Vec<u8>> {
-    Self::commit_from_db(self.db.as_ref().unwrap(), self.genesis, block)
-  }
-
-  pub(crate) fn block_hash(&self, block: u32) -> Option<[u8; 32]> {
-    Self::block_hash_from_db(self.db.as_ref().unwrap(), self.genesis, block)
+    CommitDb::get(self.db.as_ref().unwrap(), self.genesis, block)
   }
 
   pub(crate) fn commit_by_block_number(&self, block: u32) -> Option<Vec<u8>> {
-    self.commit(&self.block_hash(block)?)
-  }
-
-  pub(crate) fn block_after(db: &D, genesis: [u8; 32], block: &[u8; 32]) -> Option<[u8; 32]> {
-    db.get(Self::block_after_key(&genesis, block)).map(|bytes| bytes.try_into().unwrap())
+    CommitDb::get(
+      self.db.as_ref().unwrap(),
+      self.genesis,
+      &BlockHashDb::get(self.db.as_ref().unwrap(), self.genesis, block).unwrap(),
```

Comment:
This was `self.block_hash(block)?`. Now it's `BlockHashDb::get(...).unwrap()`. That means this function will now panic if called with a block number which hasn't been used, not return None.